easybuild/tools/toolchain/toolchain.py

```diff
@@ -363,79 +378,139 @@ def is_dep_in_toolchain_module(self, name):
-                    val = os.environ[prefix]
-                    self.log.debug("Using value of $%s as prefix for external module %s: %s", prefix, mod_name, val)
-                else:
-                    val = prefix
-                    self.log.debug("Using specified prefix for external module %s: %s", mod_name, val)
-                setvar(get_software_root_env_var_name(name), val)
+    def _simulated_load_dependency_module(self, name, version, metadata):
+        """
+        Set environment variables picked up by utility functions for dependencies specified as external modules.
+
+        @param name: software name
+        @param version: software version
+        @param metadata: dictionary with software metadata ('prefix' for software installation prefix)
+        """
+
+        self.log.debug("Defining $EB* environment variables for software named %s", name)
+
+        # define $EBROOT env var for install prefix, picked up by get_software_root
+        prefix = metadata.get('prefix')
+        if prefix is not None:
+            if prefix in os.environ:
+                val = os.environ[prefix]
+                self.log.debug("Using value of $%s as prefix for software named %s: %s", prefix, name, val)
+            else:
+                val = prefix
+                self.log.debug("Using specified prefix for software named %s: %s", name, val)
+            setvar(get_software_root_env_var_name(name), val)
 
-            # define $EBVERSION env var for software version, picked up by get_software_version
-            if version is not None:
-                setvar(get_software_version_env_var_name(name), version)
+        # define $EBVERSION env var for software version, picked up by get_software_version
+        if version is not None:
+            setvar(get_software_version_env_var_name(name), version)
 
-    def _prepare_dependencies(self):
+    def _load_toolchain_module(self, silent=False):
+        """Load toolchain module."""
+
+        tc_mod = self.det_short_module_name()
+
+        if self.dry_run:
+
+            fake_root = os.path.join(tempfile.gettempdir(), '__fake__')
+            tcmods_exists = self.modules_tool.exist([tc_mod])[0]
+
+            dry_run_msg("Loading toolchain module...\n", silent=silent)
+
+            # load toolchain module, or simulate load of toolchain components if its not available
+            if tcmods_exists:
+                self.modules_tool.load([tc_mod], silent=silent)
+                dry_run_msg("module load %s" % tc_mod, silent=silent)
+            else:
+                dry_run_msg("module load %s [SIMULATED]" % tc_mod, silent=silent)
+                # provide fake prefix so $EBROOT* gets defined
+                fake_prefix = os.path.join(fake_root, tc_mod)
+                self._simulated_load_dependency_module(self.name, self.version, {'prefix': fake_prefix})
+
+                # also simulate loads for toolchain dependencies, if required information is available
```

Comment:
shouldn't simulated the load of the toolchain deps before loading the toolchain itself?